aws/openshift/rosa-hcp-single-region/terraform/vpn/config.tf

```diff
@@ -0,0 +1,43 @@
+# Provider configuration file used only for the VPN
+# It configures the region used by the bucket storing certificates
+terraform {
+  required_version = ">= 1.6.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 5.35.0"
+    }
+  }
+
+  backend "s3" {
+    encrypt = true
+  }
+}
+provider "aws" {
+  default_tags {
+    tags = var.default_tags
+  }
+}
+
+provider "aws" {
+  region = var.s3_bucket_region
+  alias  = "aws_bucket_provider"
+  default_tags {
+    tags = var.default_tags
+  }
+}
```

Comment:
maybe I'm missing the documentation part to fully grasp the dependency on S3.
Shouldn't it be possible to have the resulting files in the state as output?

Is the dependency coming from us for easy testing etc. or is it really required?